### PR TITLE
Make OpEmitMeshTasksEXT a terminal instruction

### LINKED ISSUE #3020: Invalid control flow after EmitMeshTasksEXT()
This shader hits a bug:

```
#version 450
#extension GL_EXT_mesh_shader : require
layout(local_size_x = 4, local_size_y = 3, local_size_z = 2) in;

struct Payload
{
	float v[3];
};
taskPayloadSharedEXT Payload p;
shared float vs[24];

void main()
{
	vs[gl_LocalInvocationIndex] = 10.0;
	barrier();
	if (gl_LocalInvocationIndex < 12)
		vs[gl_LocalInvocationIndex] += vs[gl_LocalInvocationIndex + 12];
	barrier();
	if (gl_LocalInvocationIndex < 6)
		vs[gl_LocalInvocationIndex] += vs[gl_LocalInvocationIndex + 6];
	barrier();
	if (gl_LocalInvocationIndex < 3)
		vs[gl_LocalInvocationIndex] += vs[gl_LocalInvocationIndex + 3];
	barrier();

	p.v[gl_LocalInvocationIndex] = vs[gl_LocalInvocationIndex];
	if (vs[5] > 20.0)
	{
		EmitMeshTasksEXT(int(vs[4]), int(vs[6]), int(vs[8]));
	}
	else
	{
		EmitMeshTasksEXT(int(vs[6]), 10, 50u);
	}
}
```

```
              94:    6(float)   Load 93
              95:     67(int)   ConvertFToS 94
              96:      7(int)   Bitcast 95
                                EmitMeshTasksEXT 86 91 96 66(p)
                                Branch 81                                <----- *OpEmitMeshTasksEXT* must be the last instruction in a block.
              97:               Label
              98:     16(ptr)   AccessChain 11(vs) 87
              99:    6(float)   Load 98
             100:     67(int)   ConvertFToS 99
             101:      7(int)   Bitcast 100
                                EmitMeshTasksEXT 101 102 103 66(p)
                                Branch 81
```

According to SPIR-V spec, this is invalid. https://github.com/KhronosGroup/SPIRV-Registry/pull/165/files#diff-9d8cf769182417e1989cb51f487a381c2f25f4a714b010b25318dc59578e23ccR160

### PR DESCRIPTION
**Pull request by pmistryNV**
Make OpEmitMeshTasksEXT a terminal instructions

Fixes #3020

### CODE REVIEW

**Comment by pmistryNV** [comment]
@greg-lunarg please review

File: SPIRV/GlslangToSpv.cpp [hunk 1]
```diff
@@ -8559,6 +8559,8 @@ spv::Id TGlslangToSpvTraverser::createMiscOperation(glslang::TOperator op, spv::
         if(taskPayloadID)
             operands.push_back(taskPayloadID);
         builder.createNoResultOp(spv::OpEmitMeshTasksEXT, operands);
+        // Make it a terminating instruction in the current block
+        builder.createAndSetNoPredecessorBlock("post-OpEmitMeshTasksEXT");
```
Comment (greg-lunarg) [inline SPIRV/GlslangToSpv.cpp:8563]: Instead could you please create a new version of makeStatementTerminator() that takes operands and replace these statements with a call to that routine?

Then move createAndSetNoPredecessorBlock back to private?

**Review by greg-lunarg — CHANGES_REQUESTED** [review]
Please see my request. Thanks!

File: SPIRV/GlslangToSpv.cpp (hunk 1 above)
Comment (pmistryNV) [reply SPIRV/GlslangToSpv.cpp:8563]: I thought about it, but do think it's necessary? Any reason why createAndSetNoPredecessorBlock cannot be made public?

Comment (greg-lunarg) [reply SPIRV/GlslangToSpv.cpp:8563]: Its just a question of consistency and building something that might be useful to the next developer.

Comment (greg-lunarg) [reply SPIRV/GlslangToSpv.cpp:8563]: This problem is causing some issues in CI for spirv-tools, so I am going to merge it.

Please submit a follow-one PR to make the requested changes.

Comment (pmistryNV) [reply SPIRV/GlslangToSpv.cpp:8563]: https://github.com/KhronosGroup/glslang/pull/3025 to address the review comment

**Pull request merged**